x-pack/plugins/security_solution/server/lib/detection_engine/signals/threshold/bulk_create_threshold_signals.ts

```diff
@@ -48,177 +45,93 @@ interface BulkCreateThresholdSignalsParams {
 }
 
 const getTransformedHits = (
-  results: SignalSearchResponse,
+  buckets: ThresholdBucket[],
   inputIndex: string,
   startedAt: Date,
   from: Date,
   threshold: ThresholdNormalized,
   ruleId: string
 ) => {
-  if (results.aggregations == null) {
-    return [];
-  }
-  const aggParts = threshold.field.length
-    ? getThresholdAggregationParts(results.aggregations)
-    : {
-        field: null,
-        index: 0,
-        name: 'threshold_0',
-      };
-
-  if (!aggParts) {
-    return [];
-  }
-
-  const getCombinations = (buckets: TermAggregationBucket[], i: number, field: string | null) => {
-    return buckets.reduce((acc: MultiAggBucket[], bucket: TermAggregationBucket) => {
-      if (i < threshold.field.length - 1) {
-        const nextLevelIdx = i + 1;
-        const nextLevelAggParts = getThresholdAggregationParts(bucket, nextLevelIdx);
-        if (nextLevelAggParts == null) {
-          throw new Error('Unable to parse aggregation.');
-        }
-        const nextLevelPath = `['${nextLevelAggParts.name}']['buckets']`;
-        const nextBuckets = get(nextLevelPath, bucket);
-        const combinations = getCombinations(nextBuckets, nextLevelIdx, nextLevelAggParts.field);
-        combinations.forEach((val) => {
-          const el = {
-            terms: [
+  const isCompositeBucket = (bucket: ThresholdBucket): bucket is ThresholdCompositeBucket =>
+    isPlainObject(bucket.key);
+
+  return buckets.map((bucket, i) => ({
+    // In case of `terms` aggregation, `bucket.key` will be an empty string. Note that `Object.values('')` is `[]`,
+    // so the below logic works in either case (whether `terms` or `composite`).
+    _index: inputIndex,
+    _id: calculateThresholdSignalUuid(
+      ruleId,
+      startedAt,
+      threshold.field,
+      Object.values(bucket.key).sort().join(',')
+    ),
+    _source: {
+      [TIMESTAMP]: (bucket.max_timestamp as AggregationsMaxAggregate).value_as_string,
+      ...(isCompositeBucket(bucket)
```

Comment:
The 2 `isCompositeBucket` checks here make it a bit difficult to determine what the expected document format is by reading the code. It may be easier to follow if we have distinct functions for composite buckets vs the "no group by fields" case.